cartoframes/io/managers/context_manager.py

```diff
@@ -294,12 +315,38 @@ def _truncate_table(self, table_name, schema, cartodbfy):
 
     def _truncate_and_drop_add_columns(self, table_name, schema, df_columns, table_columns, cartodbfy):
         log.debug('TRUNCATE AND DROP + ADD columns table "{}"'.format(table_name))
-        query = '{regenerate}; BEGIN; {truncate}; {drop_columns}; {add_columns}; {cartodbfy}; COMMIT;'.format(
+        drop_columns = _drop_columns_query(table_name, table_columns)
+        add_columns = _add_columns_query(table_name, df_columns)
+
+        drop_add_columns = 'ALTER TABLE {table_name} {drop_columns},{add_columns};'.format(
```

Comment:
what happens when same column is dropped and added?

Reply:
It should work. It drops and adds all the columns in order to preserve the order because every new column in PG is added at the end.

Reply:
Seemed not to be a problem when performing tests on my side 🤔

Reply:
ah, OK, thanks, I was curious because now they're in the same ALTER TABLE